### PR TITLE
drivers: clk: print clock tree without recursive call

### PR DESCRIPTION
**Pull request by etienne-lms**
This series mainly replaces the implementation of `clk_print_tree()` to not use recursive calls:
commit "drivers: clk: print tree without recursion".

There are 3 other minor changes added:
* use of SLIST instead of STAILQ list in clock tree print implementation;
* assert `clk_register()` argument is not NULL;
* stub `clk_print_tree()` when `CFG_DRIVERS_CLK_PRINT_TREE` is disabled.

This is a follow up of @jenswi-linaro proposal https://github.com/OP-TEE/optee_os/pull/6481#discussion_r1404590521, applied to the clock tree.

### CODE REVIEW

**Comment by etienne-lms** [comment]
I've added a commit to remove locking clock framework main spinlock during `clk_print_tree()` execution.

**Comment by jenswi-linaro** [comment]
I'll review this once https://github.com/OP-TEE/optee_os/pull/6498 is done.

File: core/drivers/clk/clk.c [hunk 1]
```diff
@@ -386,32 +439,50 @@ static void __maybe_unused print_clock(struct clk *clk, int indent)
...
+		IMSG("-- No registered clock");
+		return;
 	}
 #endif
+
+	while (true) {
+		next = find_next_clk(parent, clk);
+		if (next) {
+			print_clk(next, indent + 1);
+			/* Enter the subtree of the next regulator */
```
Comment (jenswi-linaro) [inline core/drivers/clk/clk.c:460]: s/regulator/clock/
same below

**Comment by etienne-lms** [comment]
comment addressed

**Comment by jenswi-linaro** [comment]
`Reviewed-by: Jens Wiklander <jens.wiklander@linaro.org>`

**Comment by etienne-lms** [comment]
Fixup commit squashed and review tag applied.

**Comment by jforissier** [comment]
s/CFG_DRVIERS_CLK/CFG_DRIVERS_CLK/

**Comment by jforissier** [comment]
s/hadnling/handling/

**Comment by etienne-lms** [comment]
Comments addressed (commit messages fixed), thanks.

**Comment by etienne-lms** [comment]
I've appended a last commit "drivers: clk: print clock tree with debug trace level" for consistency with `regulator_print_tree()` implementation, if you agree.

**Review by jenswi-linaro — COMMENTED** [review]
`Reviewed-by: Jens Wiklander <jens.wiklander@linaro.org>`

**Comment by etienne-lms** [comment]
Review tags applied

**Pull request merged**